icicle/include/icicle/curves/projective.h

```diff
@@ -47,7 +47,14 @@ class Projective
     return {FF::from_montgomery(point.x), FF::from_montgomery(point.y), FF::from_montgomery(point.z)};
   }
 
+#ifdef BARRET
   static HOST_DEVICE_INLINE Projective generator() { return {Gen::gen_x, Gen::gen_y, FF::one()}; }
+#else
+  static HOST_DEVICE_INLINE Projective generator()
+  {
+    return {FF::to_montgomery(Gen::gen_x), FF::to_montgomery(Gen::gen_y), FF::one()};
```

Comment:
don't we get gen_x, gen_y from the Field as constants?
Can't you switch from 
gen_x, gen_y 
to 
gen_barret_x, gen_barret_y, gen_montgomery_x, gen_montgomery_y

Reply:
we can, it just means we need to change all the parameters in all the fields manually. This function is only called for testing so I thought this solution is good enough, but we can do it differently.